nobloat-imx-defconfig

```diff
@@ -0,0 +1,2775 @@
+# CONFIG_COMPILE_TEST is not set
+# CONFIG_USELIB is not set
+# CONFIG_GENERIC_IRQ_DEBUGFS is not set
+# CONFIG_HZ_PERIODIC is not set
+# CONFIG_NO_HZ_FULL is not set
+# CONFIG_NO_HZ is not set
+# CONFIG_PREEMPT_NONE is not set
+# CONFIG_PREEMPT_VOLUNTARY is not set
+# CONFIG_VIRT_CPU_ACCOUNTING_GEN is not set
+# CONFIG_PSI is not set
+# CONFIG_RCU_EXPERT is not set
+# CONFIG_IKHEADERS is not set
+# CONFIG_DEBUG_BLK_CGROUP is not set
+# CONFIG_CFS_BANDWIDTH is not set
+# CONFIG_RT_GROUP_SCHED is not set
+# CONFIG_CGROUP_RDMA is not set
+# CONFIG_CGROUP_FREEZER is not set
+# CONFIG_CGROUP_DEBUG is not set
+# CONFIG_CHECKPOINT_RESTORE is not set
+# CONFIG_SYSFS_DEPRECATED is not set
+# CONFIG_RELAY is not set
+# CONFIG_CC_OPTIMIZE_FOR_SIZE is not set
+# CONFIG_EXPERT is not set
```

Comment:
oh, I think you went too far here.
The idea is really to leave most of the 'kernel' parts same as defconfig - unless you explicitly need something. And you disable the 'drivers' and 'architectures/vendors' that you don't need.
We need to keep those files minimal for maintenance, the idea is not to have a minimal i.MX version but have something that's fast-enough and contains enough drivers/capabilities to test.
e.g. for intel we actually set CONFIG_EXPERT, since it unlocks a set of capabilities that are interesting if you need to alter the config manually.

So really start with removing the boards you don't need, it'll go 90% of the simplifications